ksql-functional-tests/src/main/java/io/confluent/ksql/test/model/SourceNode.java

```diff
@@ -132,4 +136,32 @@ private static ConnectSchema makeTopLevelStructNoneOptional(final Schema schema)
     schema.fields().forEach(field -> builder.field(field.name(), field.schema()));
     return (ConnectSchema) builder.build();
   }
+
+  public static class Deserializer extends JsonDeserializer<SourceNode> {
```

Comment:
the job of this deserializer is basically to distinguish between a source with no key field criteria specified, meaning the key field can be anything vs one where the key field is explicitly set to `null`, meaning the source should have no key field set.